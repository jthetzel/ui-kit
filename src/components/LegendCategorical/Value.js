import PropTypes from 'prop-types';
import styled from 'react-emotion';

const Value = styled.div`
  flex: ${({ width }) => width}%;
  font-size: 12px;
  color: #263238;
  padding-top: 5px;

  text-align: center;
`;

Value.propTypes = {
  width: PropTypes.number.isRequired
};

export default Value;
