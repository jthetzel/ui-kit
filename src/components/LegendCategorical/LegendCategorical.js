import React from 'react';
import PropTypes from 'prop-types';

import Segment from './Segment';
import Value from './Value';

// styled
import Container from './Container';

/**
 * @component
 */

const LegendCategorical = ({ values }) => {
  if (!values.length) {
    return null;
  }

  const width = 100 / (values.length);

  return (
    <Container>
      {
        values.map(({ id, color }) =>
          <Segment key={id} width={width} color={color} />)
      }
      {
        values.map(({ id, title }) => (
          <Value
            key={id}
            width={width}
          >
            {title}
          </Value>
        ))
      }
    </Container>
  );
};

LegendCategorical.propTypes = {
  values: PropTypes.arrayOf(PropTypes.shape({
    id: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
    color: PropTypes.string,
    title: PropTypes.node
  })).isRequired
};

export default LegendCategorical;
