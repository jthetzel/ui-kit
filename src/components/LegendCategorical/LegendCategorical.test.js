import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import LegendCategorical from './';

const values = [
  { id: 1, color: '#111111', title: 0 },
  { id: 2, color: '#333333', title: 100 },
  { id: 3, color: '#666666', title: 300 },
  { id: 4, color: '#999999', title: 600 },
  { id: 5, color: '#AAAAAA', title: 900 }
];

test('Legend does not crash', () => {
  const tree = shallow(<LegendCategorical values={values} />);
  expect(toJson(tree)).toMatchSnapshot();
});

test('Legend got an empty array', () => {
  const tree = shallow(<LegendCategorical values={[]} />);
  expect(toJson(tree)).toMatchSnapshot();
});
