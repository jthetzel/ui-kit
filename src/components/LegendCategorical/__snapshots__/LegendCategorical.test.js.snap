// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Legend does not crash 1`] = `
<Styled(div)>
  <Styled(div)
    color="#111111"
    key="1"
    width={20}
  />
  <Styled(div)
    color="#333333"
    key="2"
    width={20}
  />
  <Styled(div)
    color="#666666"
    key="3"
    width={20}
  />
  <Styled(div)
    color="#999999"
    key="4"
    width={20}
  />
  <Styled(div)
    color="#AAAAAA"
    key="5"
    width={20}
  />
  <Styled(div)
    key="1"
    width={20}
  >
    0
  </Styled(div)>
  <Styled(div)
    key="2"
    width={20}
  >
    100
  </Styled(div)>
  <Styled(div)
    key="3"
    width={20}
  >
    300
  </Styled(div)>
  <Styled(div)
    key="4"
    width={20}
  >
    600
  </Styled(div)>
  <Styled(div)
    key="5"
    width={20}
  >
    900
  </Styled(div)>
</Styled(div)>
`;

exports[`Legend got an empty array 1`] = `""`;
